src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -422,4 +438,38 @@ function AppleHealthMenuItem(props: AppleHealthMenuItemProps) {
             </div>
         }
     </div>;
+}
+
+function EnvironmentalMenuItem(props: { settings: DataCollectionSettings | null, participantInfo: ParticipantInfo | null, postalCodeSurveyName: string }){
+    let action = () => {
+        MyDataHelps.startSurvey(props.postalCodeSurveyName);
+    }
+
+    const postalCodeCustomFields = Object.keys(props.participantInfo ? props.participantInfo.customFields : {}).filter( (k) => k.endsWith('PostalCode'));
+    const postalCodes = [...postalCodeCustomFields.map((cf) => props.participantInfo?.customFields[cf]), props.participantInfo?.demographics.postalCode];
+    var postalCode = postalCodes.filter((p) => !!p).slice(0,3).join(', ');
+    if(postalCodes.length > 3) {
```

Comment:
If 3 ends up being too long, I think 2 + ", ..." would be fine as well.  I suspect most projects don't go above 2.

Alternatively, maybe you could join them all and update the `mdhui-connect-devices-menu-connect` style class to use text overflow ellipsis.  I haven't tinkered with that though, so it may be more trouble than it is worth.  😃

Reply:
Yeah, I figured most wouldn't get beyond 2, but based on the length of the next longest thing which seems to show up in this area which is "Download MyDataHelps", 3 seemed to fit fine.  I figured it was an unlikely edge case in general, but wanted to cover it just in case.  I didn't want to mess with the css class just to have this sort of decision be localized to this specific menu item.